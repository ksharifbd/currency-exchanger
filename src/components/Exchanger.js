import React from 'react';
import Box from '@material-ui/core/Box';
import styled from '@material-ui/core/styles/styled';
import ArrowRightAltIcon from '@material-ui/icons/ArrowRightAlt';
import ExchangeEnabler from './ExchangeEnabler';

const Exchanger = () => {
  const ExchangerWrapper = styled(Box)({
    display: 'flex',
    flexWrap: 'nowrap',
    justifyContent: 'space-evenly',
    alignItems: 'center',
  });

  return (
    <ExchangerWrapper>
      <ExchangeEnabler />
      <ArrowRightAltIcon />
      <ExchangeEnabler />
    </ExchangerWrapper>
  );
};

export default Exchanger;
