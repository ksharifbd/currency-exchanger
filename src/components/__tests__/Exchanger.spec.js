import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Exchanger from '../Exchanger';

describe('Exchanger', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  const mockFn = jest.fn();

  const defaultProps = {
    currencyFromSymbol: '$',
    currencyToSymbol: '$',
    selectedCurrencyFrom: 'USD',
    selectedCurrencyTo: 'USD',
    onExchange: mockFn,
    onFromSelectChange: mockFn,
    onToSelectChange: mockFn,
    conversionRate: 5,
    inputCurrencyBalance: 10,
    outputCurrencyBalance: 50,
    isEverythingUp: false,
  };

  const tree = (props = defaultProps) => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    return shallow(<Exchanger {...props} />);
  };

  it('should render basic setup correctly', () => {
    expect(tree()).toMatchSnapshot();
  });

  it('should not call onSubmit callback on clicking disabled button', () => {
    tree()
      .find({ type: 'submit' })
      .simulate('click');

    expect(mockFn).not.toHaveBeenCalled();
  });
});
