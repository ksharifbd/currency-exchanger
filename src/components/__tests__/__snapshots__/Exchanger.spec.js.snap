// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchanger should render basic setup correctly 1`] = `
<Styled(MuiBox)>
  <form
    onSubmit={[Function]}
  >
    <Styled(Styled(MuiBox))
      mb={3}
    >
      <ExchangeInput
        currencySymbol="please provide USD, GBP, or EUR"
        currencyValue=""
        hasInputError={true}
        inputCurrencyBalance="10"
        inputHelperText="Same currency exchange is not allowed"
        onCurrencySelect={[Function]}
        onCurrencyValueChange={[Function]}
        selectedCurrency="USD"
      />
      <ArrowRightAltIcon />
      <ExchangeOutput
        convertedValue=""
        currencySymbol="please provide USD, GBP, or EUR"
        onCurrencySelect={[Function]}
        outputCurrencyBalance="50"
        outputCurrencyExchangeRate="5"
        selectedInputCurrency="USD"
        selectedOutputCurrency="USD"
      />
    </Styled(Styled(MuiBox))>
    <Styled(Styled(MuiBox))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        disabled={true}
        type="submit"
        variant="contained"
      >
        Exchange
      </WithStyles(ForwardRef(Button))>
    </Styled(Styled(MuiBox))>
  </form>
</Styled(MuiBox)>
`;
